frontend/ui/widget/inputtext.lua

```diff
@@ -297,13 +297,19 @@ end
 local function initDPadEvents()
     if Device:hasDPad() then
         function InputText:onFocus()
-            -- Event called by the focusmanager
+            -- Event sent by focusmanager
             if self.parent.onSwitchFocus then
                 self.parent:onSwitchFocus(self)
             elseif (Device:hasKeyboard() or Device:hasScreenKB()) and G_reader_settings:isFalse("virtual_keyboard_enabled") then
                 do end -- luacheck: ignore 541
             else
-                self:onShowKeyboard()
+                if not self:isKeyboardVisible() then
+                    -- Don't break InputDialog:toggleKeyboard...
+                    local skip_vk = self.parent and self.parent._manual_vk_toggle
+                    if not skip_vk then
+                        self:onShowKeyboard()
+                    end
+                end
```

Comment:
It feels (not sure) this could be related to a dirty suggestion I gave previously and ended up with having `skip_first_show_keyboard` in InputDialog (dirty, because if the upper widget doesn't explicitely call onShowKeyboard(), and it's supposed to happen on user action, the first user action will then be `skip`ped/ignored.
So, if it is somehow related, feel free to clean my dirty suggestion.

Reply:
I maintain that, even with that dragon slain, that button still makes little sense on NT, because, the second you remove focus from a field the VK disappears and as soon as one re-focuses on an inputtext, is back. Just sayin'

furthermore the `elseif` would need to change as well as, the button won't pop up the keyboard (which would need to be an InputDialogue one not an InputText one) otherwise `skip_first_show_keyboard` was in fact pointless and should go.

Reply:
> So, if it is somehow related, feel free to clean my dirty suggestion.

Nope, unrelated.

That one's because it's, for some mysterious reason, the caller's responsibility to call inputdialog's onShowKeyboard; instead of just having InputDialog do it on its own via onShow; so basically anything that shows an InputDialog does

```lua
local dialog = InputDialog:new{...}
UIManager:show(dialog)
diualog:onShowKeyboard()
```

Which is mad, granted, but unrelated to our current woes ;p.

Reply:
I think the idea is probably that some may decide not to show by default, although I'm not really sure why that wouldn't just be a flag.